provider/provider.go

```diff
@@ -154,6 +155,11 @@ func (p *dockerNativeProvider) Check(ctx context.Context, req *rpc.CheckRequest)
 				return nil, err
 			}
 		}
+
+		if _, err := marshalCachedImages(inputs["build"]); err != nil {
+			err = p.host.Log(ctx, diag.Error, urn, msg)
+			return nil, err
+		}
```

Comment:
yeah! I love verifying this early on during preview.

Can we write a PR description to this effect?